opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -266,13 +266,18 @@ class OTELResourceDetector(ResourceDetector):
     def detect(self) -> "Resource":
         env_resources_items = os.environ.get(OTEL_RESOURCE_ATTRIBUTES)
         env_resource_map = {}
+
         if env_resources_items:
-            env_resource_map = {
-                key.strip(): value.strip()
-                for key, value in (
-                    item.split("=") for item in env_resources_items.split(",")
-                )
-            }
+            for item in env_resources_items.split(","):
+                if item.find("=") == -1:
```

Comment:
nit: use `in` it is faster and reads like english

Reply:
Another option is a try/except block around the tuple unpack which works for more possible parsing errors. E.g. if there are multiple `=` characters you get `ValueError: too many values to unpack`

Reply:
Thanks all! I think those are great suggestion, and definitely @aabmass I often forget it's better to ask for forgiveness than for permission in Python 🙂.

I'm going to catch only `ValueError` because that should be all that occurs here I think?

Reply:
Since you are passing the `maxsplit` value here `key, value = item.split("=", 1)` it shouldn't raise that error when there are multiple `=` characters.

Reply:
Oh that's a good point... I guess I will remove the `maxsplit` error so that we can skip the `foo=bar=baz` case that Aaron mentioned below and was probably his intention?